sources/commands/option_cd.c

```diff
@@ -6,19 +6,62 @@
 /*   By: inikulin <inikulin@student.42berlin.de>    +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2024/09/30 00:08:15 by inikulin          #+#    #+#             */
-/*   Updated: 2024/10/12 01:10:27 by inikulin         ###   ########.fr       */
+/*   Updated: 2024/10/23 00:20:53 by inikulin         ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "commands.h"
 
+static char	*get_checked_path(t_treenode *node, char *pwd, t_mapss *envvars)
+{
+	char	*tgt;
+
+	if (!node->child)
+		return (ft_strdup(ft_mapss_get(envvars, "HOME")));
```

Comment:
Should we consider scenario if the HOME variable is missing? then the get_checked_path function  will not be able to return the correct path for calling chdir. 

maybe to add a check?

```
char *home = ft_mapss_get(envvars, "HOME");

if (!home || *home == '\0')
{
    printf("cd: HOME not set\n");
    return 1;
}
```

Reply:
Good point, thank you